### ISSUE TITLE
Building a hidden Posts feature in Astro using the new Strapi v5 Drafts & Publish feature 

### ISSUE THREAD

**Issue by noahflk** [strapi/community-content#1523]
### What is your article idea?

This article will be a tutorial and showcase of the new Strapi V5 [Drafts & Publish](https://docs-next.strapi.io/user-docs/content-manager/saving-and-publishing-content) feature.

We will build a simple blog in Astro where you can publish hidden draft posts that aren't publicly accessible. You can use this to present upcoming blog posts to your fellow team members for review directly on your blog - before they're published.

**Sections**

- Introduction
   - Show our Tech Stack: Strapi, Astro and Tailwind
   - Benefits of publishing drafts
   - Requirements (Node.js)
-  Setting up strapi
   - Initial setup
   - Content mode
   - Configure API access
- Getting started with Astro
    - Bootstrapping project
    - Adding Tailwind, design skelleton
    - Accessing Strapi API
- Draft support
   - Draft explanation
   - Accessing drafts through the API
   - Rendering drafts
- Conclusion

### What are the objectives of your article?

Promoting the Strapi v5 Release with the new [Drafts & Publish](https://docs-next.strapi.io/user-docs/content-manager/saving-and-publishing-content) feature and show how you could use it in an Astro Strapi blog.

### What is your expertise as a developer or writer?

Intermediate

### What type of post is this?

Tutorial

### Terms & Conditions

- [X] I have read the Write for the Community program guidelines.

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Hi @noahflk ,

It is great you want to use the Strapi 5 draft and publish feature but I am not sure what you mean by hidden post features. What are we building?

**Comment by noahflk** [comment]
@Theodore-Kelechukwu-Onyejiaku 

> This sounds interesting! Meaning that posts that are drafts and published will be gotten using the status and published query?

Yes exactly!

> Is there a better title you might suggest other than "Building a hidden posts feature in Astro..."?

This sounds good to me. Should I get started?

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Please suggest a better title.

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
How about "Build a note keeper with Strapi Draft and Publish"?

Or do you have a better suggestion please?

**Comment by noahflk** [comment]
What about "Building a Blog Post Preview feature with Strapi Draft and Publish"

I think the blogging aspect works a bit better as an example. What do you think?

@Theodore-Kelechukwu-Onyejiaku

**Comment by noahflk** [comment]
Hello @Theodore-Kelechukwu-Onyejiaku here is my draft: https://hackmd.io/@noahflk/S1NMXwN30

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Hi @noahflk ,

Thank you. Could you please provide me the following details on discord:
* Full Name
* Photo
* Bio
* Email 
* Twitter Handle

Thank you for your contribution to the "Write for the Community" program!

**Comment by noahflk** [comment]
@Theodore-Kelechukwu-Onyejiaku just sent it on Discord DM

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Thank you!

Just a few requests. I am hoping we can make this intermediate. How about we include Strapi customization or backend customization by introducing the new Strapi Document Service API functions: 

* count()
* publish()
* unpublish()
* discardDraft()

That way, we could have these functions integrated into the app in your work. So a user can can count documents published or in draft, publish a draft, unpublish a draft and discard a draft.

Please let me know if you are willing to incorporate this.

**Comment by noahflk** [comment]
Thank you @Theodore-Kelechukwu-Onyejiaku 

I am happy to incorporate it. 

Is this what you have in mind:
- Each post gets a publish/unpublish/discard button?

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Hi @noahflk ,

Yes that is what I have in mind. And this time, it should be a form of a custom controller(s) which should make use of the Document Service to call these new functions. 

Please let me know if you understand. Thank you!

**Comment by noahflk** [comment]
@Theodore-Kelechukwu-Onyejiaku got it, will deliver asap

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Thank you!

**Comment by noahflk** [comment]
@Theodore-Kelechukwu-Onyejiaku 

I have submitted the updated version. it is available under the same URL: https://hackmd.io/@noahflk/S1NMXwN30

**Comment by Theodore-Kelechukwu-Onyejiaku** [comment]
Amazing work @noahflk, I am impressed by the simplicity and use of the new features of the Draft and Publish feature. Nice job!

One last thing please.

A post entry could have the published and draft versions. Is there a need creating another entry to show that a particular content 
is a draft or published content? 

For example, an entry A has a published and draft version. And on the client side, we could fetch all published versions of any entry, perhaps using a route called `/blog/published`. And we could show all drafts using `/blog/drafts. This feature is common on Medium.com. This way, we could also call these new functions on them. 

I have doubts that the one presented in your draft is the proper way to show the Draft and Publish feature in Strapi 5. 

Please let me know what you think.

I also realized you didn't add demos/GIFs to show the `publish()`, `unpublish()` and `discardDraft()` functions. I will add them myself! 💪

Great job! Thank you very much.

**Comment by noahflk** [comment]
Thanks @Theodore-Kelechukwu-Onyejiaku 

I have updated the blog post with the following

- Published posts are accessible at `/blog/SLUG`
- Draft posts are accessible `/blog/draft/SLUG`

This takes into account the fact that a single post/entry can have both a publsihed and draft version. I chose not to provide a list of draft posts at `/blog/draft` since draft posts should only be accessible to those that know the URL. Since you wouldn't want every reader to browse your drafts.

Let me know if that works for you.
